### PR TITLE
DOC: more pd.to_datetime examples

### PR DESCRIPTION
**Pull request by jreback**
xref #12996

### CODE REVIEW

**Comment by jreback** [comment]
@jorisvandenbossche

File: pandas/tseries/tools.py [hunk 1]
```diff
@@ -257,12 +257,33 @@ def to_datetime(arg, errors='raise', dayfirst=False, yearfirst=False,
...
+
+    >>> s.head()
+    0    3/11/2000
+    1    3/12/2000
+    2    3/13/2000
+    3    3/11/2000
+    4    3/12/2000
+    dtype: object
+
+    >>> %timeit pd.to_datetime(s,infer_datetime_format=True)
```
Comment (jreback) [inline pandas/tseries/tools.py]: is this legit to do here?

**Comment by jreback** [comment]
also added #12974 fix here.

**Comment by jorisvandenbossche** [comment]
@jreback Can you please try to not change the commit too much during the merge? This makes reviewing PRs / seeing what has been merged a lot more difficult using github (as it seems the 'Files changed' is quite different from the actual merged commit).

(of course, small fixup changes are not a problem, certainly for PRs of other contributors to not have them endlessly adapt minor things)

**Comment by jreback** [comment]
@jorisvandenbossche I normally don't. but ended up going on a tangent with some other things (somewhat unrelated).

**Comment by jorisvandenbossche** [comment]
That's OK! (I just noticed it for this one)

**Pull request closed without merging**